index.js

```diff
@@ -545,7 +546,7 @@ function parse (args, opts) {
       // if the value is an inner object and we have dot-notation
       // enabled, treat inner objects in config the same as
       // heavily nested dot notations (foo.bar.apple).
-      if (typeof value === 'object' && value !== null && !Array.isArray(value) && configuration['dot-notation']) {
+      if (typeof value === 'object' && value !== null && !Array.isArray(value) && configuration['config-dot-notation']) {
```

Comment:
This logic seems wrong to me, which I believe is why you've had to explicitly turn on `config-dot-notation` to get tests passing.

What I believe we want, when `config-dot-notation` is `false` is to still run `setConfigObject` (which takes care of merging the config file with our arguments.

It's a specific edge-case you want to turn off, which is how someone handles a config file that looks like this:

```
{
  "foo.com": {
     "hello": "world"
  }
}
```